lib/roast/dsl/cogs/chat.rb

```diff
@@ -40,40 +40,19 @@ def initialize(response)
         end
 
         class Config < Cog::Config
-          #: (?String?) -> String?
-          def api_key(key = nil)
-            @values[:api_key] ||= key || ENV["OPENAI_API_KEY"]
-          end
-
-          #: (?String?) -> String?
-          def base_url(url = nil)
-            @values[:base_url] ||= url || ENV["OPENAI_API_BASE_URL"]
-          end
-
-          #: (?String?) -> String?
-          def model(model_name = nil)
-            @values[:model] ||= model_name
-          end
-
-          #: (?Symbol?) -> Symbol?
-          def provider(provider_name = nil)
-            @values[:provider] ||= provider_name
-          end
-
-          #: (?bool?) -> bool?
-          def assume_model_exists(assume_model_exists = nil)
-            @values[:assume_model_exists] ||= assume_model_exists || false
-          end
+          field :model, String
```

Comment:
Strong opinion: the type should not be part of the runtime definition. We can handle this with a Sorbet compiler down the line, much like how `attr_accessor` is already handled.